dataland-backend/src/main/kotlin/org/dataland/datalandbackend/services/DataManager.kt

```diff
@@ -66,45 +75,125 @@ class DataManager(
+        sendMessageToQueue(dataId, "New data - QA necessary", correlationId, "upload_queue")
         return dataId
     }
 
-    private fun storeDataSet(
+    private fun sendMessageToQueue(dataId: String, type: String, correlationId: String, messageQueue: String) {
+        try {
+            cloudEventMessageHandler.buildCEMessageAndSendToQueue(
+                dataId, type, correlationId,
+                messageQueue,
+            )
+        } catch (exception: AmqpException) {
+            val internalMessage = "Error sending message to $messageQueue." +
+                " Received AmqpException with message: ${exception.message}. Correlation ID: $correlationId."
+            logger.error(internalMessage)
+            throw AmqpException(internalMessage, exception)
+        }
+    }
+
+    /**
+     * Method that listens to the qa_queue and updates the metadata information after successful qa process
+     * @param message is the message delivered on the message queue
+     */
+    @RabbitListener(queues = ["qa_queue"])
+    fun updateMetaDataAfterQA(message: Message) {
+        val dataId = cloudEventMessageHandler.bodyToString(message)
+        val correlationId = message.messageProperties.headers["cloudEvents:id"].toString()
+        if (!dataId.isNullOrEmpty()) {
+            val metaInformation = metaDataManager.getDataMetaInformationByDataId(dataId)
+            metaDataManager.storeDataMetaInformation(metaInformation.copy(qaStatus = QAStatus.Accepted))
+            logger.info(
+                "Received quality assurance for data upload with DataId: $dataId with Correlation Id: $correlationId",
+            )
+        } else {
+            val internalMessage = "Error updating metadata data. Correlation ID: $correlationId"
+            logger.error(internalMessage)
+            throw InternalServerErrorApiException(
+                "Update of meta data failed", "The update of the metadataset failed",
+                internalMessage,
+            )
+        }
+    }
+
+    /**
+     * This method retrieves data from the temporal storage
+     * @param dataId is the identifier for which all stored data entries in the temporary storage are filtered
+     */
+    fun selectDataSetForInternalStorage(dataId: String): String {
+        val rawValue = dataInformationHashMap.getOrElse(dataId) {
+            throw ResourceNotFoundApiException(
+                "Non-persisted data ID not found",
+                "Dataland does not know the non-persisted data id $dataId",
+            )
+        }
+        return objectMapper.writeValueAsString(rawValue)
+    }
+
+    /**
+     * Method to temporarily store a data set via the hash map and send a notification to the storage_queue
+     * @param storableDataSet The data set to store
+     * @param companyName The name of the company corresponding to the data set to store
+     * @param correlationId
+     */
+    fun storeDataSet(
```

Comment:
I think we should reflect what is really going on: 
The backend does two things: It receives data and tells the world (via MQ) that it has done so. 
Then - the backend listens for random services to proclaim that QA is done. 

Currently, the method names don't correspond to this - e.g. `StoreDataSet` is more likely `SendMessageDatasetReceived` or the like.

Please go through this whole file and adapt accordingly. Also update Docstrings to reflect what methods do.

Reply:
Updated the method names and docstrings to be more aligning to what they actual do.

Reply:
ok